lieer/local.py

```diff
@@ -303,16 +310,23 @@ def messages_to_gids (self, msgs):
           print ("'%s' is not in this repository, ignoring." % fname)
         else:
           # get gmail id
-          gid = os.path.basename (fname).split (':')[0]
+          gid = self.__filename_to_gid__ (os.path.basename (fname))
           gids.append (gid)
           messages.append (m)
 
     return (messages, gids)
 
 
+  def __filename_to_gid__ (self, fname):
+    return fname.split (':')[0].split('.')[0]
```

Comment:
I think it would be better to use the full configured extension, perhaps using `rfind`, presumably `:` can be added as part of the search key to avoid two splits/rfinds and make the splitting unambigious. This will fail now if the configured extension contains `.` or `:`. We also need to add a section to the README about changing this option after synchronization similar to a few of the other options. Essentially you need to delete existing files and re-sync everything unless you manually rename all the files (since we are not providing a migration).

Reply:
I thought about using the full extension, but there would be problems if that extension was changed without renaming mail synchronized with a previous extension.

I agree though that the `--file-extension` option should reject arguments that contain `.` or `:`. I hadn't thought of that.

My idea was that migration is a separate problem, solved either with a full resync as you suggest, or with a simple shell script.

Reply:
OK. You should, of course, edit my patch as you see fit.